src/Sentry.Unity/Integrations/UnityErrorLogException.cs

```diff
@@ -0,0 +1,170 @@
+    /// </summary>
+    /// <remarks>
+    /// <see cref="Application.logMessageReceived"/>
+    /// </remarks>
+    internal class UnityErrorLogException : Exception
+    {
+        internal static readonly string ExceptionType = "LogError";
+
+        private readonly string _logString = string.Empty;
+        private readonly string _logStackTrace = string.Empty;
+
+        private readonly SentryOptions? _options;
+
+        public UnityErrorLogException(string logString, string logStackTrace, SentryOptions? options)
+        {
+            _logString = logString;
+            _logStackTrace = logStackTrace;
+            _options = options;
+        }
+
+        internal UnityErrorLogException(string logString, string logStackTrace)
+        {
+            _logString = logString;
+            _logStackTrace = logStackTrace;
+        }
+
+        internal UnityErrorLogException() : base() { }
+
+        private UnityErrorLogException(string message) : base(message) { }
+
+        private UnityErrorLogException(string message, Exception innerException) : base(message, innerException) { }
+
+        public SentryException ToSentryException()
+        {
+            var frames = ParseStackTrace(_logStackTrace);
+            frames.Reverse();
+
+            var stacktrace = new SentryStackTrace { Frames = frames };
+
+            return new SentryException
+            {
+                Stacktrace = stacktrace,
+                Type = ExceptionType,
+                Value = _logString,
+                Mechanism = new Mechanism
+                {
+                    Handled = true,
+                    Type = "unity.log"
+                }
+            };
+        }
+
+        private const string AtFileMarker = " (at ";
+
+        private List<SentryStackFrame> ParseStackTrace(string stackTrace)
+        {
+            // Example: Sentry.Unity.Integrations.UnityLogHandlerIntegration:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[]) (at UnityLogHandlerIntegration.cs:89)
+            // This follows the following format:
+            // Module.Class.Method[.Invoke] (arguments) (at filepath:linenumber)
+            // The ':linenumber' is optional and will be omitted in builds
+
+            var frames = new List<SentryStackFrame>();
+            var stackList = stackTrace.Split('\n');
+
+            foreach (var line in stackList)
+            {
+                var item = line.TrimEnd('\r');
+                if (string.IsNullOrEmpty(item))
+                {
+                    continue;
+                }
+
+                var frame = ParseStackFrame(item);
+                if (_options is not null)
+                {
+                    frame.ConfigureAppFrame(_options);
+                }
+                frames.Add(frame);
+            }
+
+            return frames;
+        }
+
+        private static SentryStackFrame ParseStackFrame(string stackFrameLine)
```

Comment:
Personally I'd have preferred regex as that would have been easier to read to _me_. But AFAICT, the manual parsing seems correct